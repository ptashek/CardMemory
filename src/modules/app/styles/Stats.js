export default (theme) => ({
  stats: {
    margin: theme.spacing(2),
    padding: theme.spacing(1),
  },
});
