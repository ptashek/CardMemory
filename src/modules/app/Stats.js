// @flow
import type { ComponentType } from 'react';
import type { State, Dispatch } from '../../hooks/useAppState';
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import AppStateContext from './AppStateContext';
import styles from './styles/Stats';

const useStyles = makeStyles(styles);

const Stats: ComponentType<{}> = () => {
  const classes = useStyles();
  // eslint-disable-next-line no-unused-vars
  const [state, _, pairCount]: [State, Dispatch, number] = React.useContext(AppStateContext);

  // prettier-ignore
  return (
    <Typography variant="h6" className={classes.stats}>
      Click each card to reveal it. When a pair is matched, it will remain visible until the game finishes.<br />
      Unmatched pairs will be hidden after a short delay.<p/>
      You have solved {state.solved.size} out of {pairCount} pairs in {Math.floor(state.moves)} moves
    </Typography>
  );
};

export default React.memo<{}>(Stats);
