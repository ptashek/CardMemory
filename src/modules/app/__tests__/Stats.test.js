import React from 'react';
import Stats from '../Stats';

describe('<Stats />', () => {
  test('matches snapshot', () => {
    const { container } = renderWithTheme(<Stats />);
    expect(container).toMatchSnapshot();
  });
});
