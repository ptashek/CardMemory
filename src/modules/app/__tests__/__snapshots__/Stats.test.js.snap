// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stats /> matches snapshot 1`] = `
<div>
  <h6
    class="MuiTypography-root makeStyles-stats-1 MuiTypography-h6"
  >
    Click each card to reveal it. When a pair is matched, it will remain visible until the game finishes. Only two unmatched cards can be revealed at any given time.
  </h6>
</div>
`;
